proxy/contracts/mainnet/DepositBoxes/DepositBoxERC20.sol

```diff
@@ -288,12 +510,137 @@ contract DepositBoxERC20 is DepositBox, IDepositBoxERC20 {
+                }
+            }
+        }
+    }
+
+    /**
+     * @dev Get amount of addresses that are added to the whitelist
+     */
+    function getTrustedReceiversAmount(bytes32 schainHash) external view override returns (uint256) {
+        return _delayConfig[schainHash].trustedReceivers.length();
+    }
+
+    /**
+     * @dev Get i-th address of the whitelist
+     */
+    function getTrustedReceiver(string calldata schainName, uint256 index) external view override returns (address) {
+        return _delayConfig[_schainHash(schainName)].trustedReceivers.at(index);
+    }
+
+    /**
+     * @dev Get amount of tokens that are considered as a big transfer
+     */
+    function getBigTransferThreshold(bytes32 schainHash, address token) external view override returns (uint256) {
+        return _delayConfig[schainHash].bigTransferThreshold[token];
+    }
+
+    /**
+     * @dev Get time delay of big transfers
+     */
+    function getTimeDelay(bytes32 schainHash) external view override returns (uint256) {
+        return _delayConfig[schainHash].transferDelay;
+    }
+
+    /**
+     * @dev Get duration of an arbitrage
+     */
+    function getArbitrageDuration(bytes32 schainHash) external view override returns (uint256) {
+        return _delayConfig[schainHash].arbitrageDuration;
+    }
+
+    /**
+     * @dev Retrive tokens that were unlocked after delay for specified receiver
+     */
+    function retrieveFor(address receiver) public override {
+        uint256 transfersAmount = MathUpgradeable.min(
+            delayedTransfersByReceiver[receiver].length(),
+            _QUEUE_PROCESSING_LIMIT
+        );
+
+        uint256 currentIndex = 0;
+        for (uint256 i = 0; i < transfersAmount; ++i) {
+            uint256 transferId = uint256(delayedTransfersByReceiver[receiver].at(currentIndex));
+            DelayedTransfer memory transfer = delayedTransfers[transferId];            
+            ++currentIndex;
+
+            if (transfer.status != DelayedTransferStatus.COMPLETED) {
+                if (block.timestamp < transfer.untilTimestamp) {
+                    // disable detector untill slither fixes false positive
+                    // https://github.com/crytic/slither/issues/778
+                    // slither-disable-next-line incorrect-equality
+                    if (transfer.status == DelayedTransferStatus.DELAYED) {
+                        break;
+                    } else {
+                        // status is ARBITRAGE
+                        continue;
+                    }
+                } else {
+                    // it's time to unlock
+                    if (currentIndex == 1) {
+                        --currentIndex;
+                        _removeOldestDelayedTransfer(receiver);
+                    } else {
+                        delayedTransfers[transferId].status = DelayedTransferStatus.COMPLETED;
```

Comment:
Is this body will be executed whenever?

Loop is starting from the lowest "untilTimestamp" so if there are some delayed transfers which completed they are would be in beginning positions so a body above would be executed

Reply:
Yes it will.

For example in this test case:
https://github.com/skalenetwork/IMA/blob/feature/delay/proxy/test/DepositBoxERC20.ts#L598

If 2 transfers were delayed and first was arbitraged and second exceeded a delay time limit.